packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -266,44 +287,68 @@ export function useConnections({
     }
   }
 
-  async function onConnectSuccess(
-    connectionInfo: ConnectionInfo,
-    dataService: DataService
-  ) {
-    // After connecting and the UI is updated we notify the rest of Compass.
-    try {
-      onConnected(connectionInfo, dataService);
-
-      // if a connection has been saved already we only want to update the lastUsed
-      // attribute, otherwise we are going to save the entire connection info.
-      const connectionInfoToBeSaved =
-        (await connectionStorage.load(connectionInfo.id)) ?? connectionInfo;
-
-      await saveConnectionInfo({
-        ...cloneDeep(connectionInfoToBeSaved),
-        lastUsed: new Date(),
+  async function removeConnection(connectionInfo: ConnectionInfo) {
+    await connectionStorage.delete(connectionInfo);
+    dispatch({
+      type: 'set-connections',
+      connections: connections.filter((conn) => conn.id !== connectionInfo.id),
+    });
+    if (activeConnectionId === connectionInfo.id) {
+      const nextActiveConnection = createNewConnectionInfo();
+      dispatch({
+        type: 'set-active-connection',
+        connectionInfo: nextActiveConnection,
       });
-    } catch (err) {
-      debug(
-        `error occurred connection with id ${connectionInfo.id || ''}: ${
-          (err as Error).message
-        }`
-      );
     }
   }
 
-  useEffect(() => {
-    if (
-      isConnected &&
-      connectedConnectionInfo.current &&
-      connectedDataService.current
-    ) {
-      void onConnectSuccess(
-        connectedConnectionInfo.current,
-        connectedDataService.current
-      );
-    }
-  }, [isConnected, onConnected]);
+  const onConnectSuccess = useCallback(
+    async (connectionInfo: ConnectionInfo, dataService: DataService) => {
+      // After connecting and the UI is updated we notify the rest of Compass.
```

Comment:
I think this comment can go away now that we've made this connect success not occur on a `useEffect`?